### PR TITLE
Download Temporal Server with UI

### PR DESCRIPTION
**Pull request by stevekinney**
Downloads and unpacks Temporal server when install dependencies.

Adds scripts to start server and UI at the same time.

### CODE REVIEW

File: download-temporal-cli.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  console.error(kleur.bgRed('Error:'), kleur.red(error));
+  callback();
+  process.exit(2);
+};
+
+const destinationDirectory = './bin';
+const destination = join(destinationDirectory, 'cli');
+
+const latestRelease = await fetch(
+  'https://api.github.com/repos/temporalio/cli/releases/latest',
```
Comment (feedmeapples) [inline download-temporal-cli.js:23]: in case this hits github rate limit we can swap with https://github.com/temporalio/cloudflare-release-manager#urls

Comment (feedmeapples) [reply download-temporal-cli.js:23]: might make few changes though so temporal.download returns the archive itself instead of just a download link

Comment (stevekinney) [reply download-temporal-cli.js:23]: Yea, I think we should _totally_ do that but maybe I wait until you're done making breaking changes? Ideally, we shouldn't be running `pnpm install` too often anyway.

Comment (bergundy) [reply download-temporal-cli.js:24]: You should use our temporal.download CDN.
In TS we've been bit by GH rate limits before.

File: README.md [hunk 2]
```diff
@@ -22,6 +43,15 @@ pnpm run build:local
 pnpm run preview:local
 ```
 
+### Using Docker
+
+If you're running the development version of the UI and you want to point it at the `docker-compose` version of Temporal, you can run this command:
+
+```
+pnpn run build:docker
+pnpn run preview:docker
```
Comment (rossedfort) [inline README.md:52]: Nit: `pnpm -> pnpm`

**Comment by stevekinney** [comment]
Currently waiting on https://github.com/temporalio/cloudflare-release-manager/pull/7

File: download-temporal-cli.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { join } from 'path';
```
Comment (rossedfort) [inline download-temporal-cli.js:1]: Should we put this in a `/bin`, or `/scripts` directory?

Comment (stevekinney) [reply download-temporal-cli.js:1]: `/bin` is ignored, so not there. Maybe if we ever have more than one, we can move it into a `scripts` directory.

**Review by rossedfort — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -124,14 +132,16 @@
...
     "ts-node": "^10.2.1",
     "ts-proto": "^1.82.5",
     "tslib": "^2.3.1",
     "typescript": "^4.4.2",
     "vite": "^4.0.2",
     "vite-node": "^0.23.2",
     "vitest": "^0.15.2",
-    "webpack": "^5.73.0"
+    "webpack": "^5.73.0",
+    "zlib": "^1.0.5"
```
Comment (GiantRobots) [inline package.json:144]: https://github.com/kkaefer/DEPRECATED-node-zlib

File: download-temporal-cli.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+const reportError = (error, exitCode = 1, callback) => {
+  console.error(kleur.bgRed('Error:'), kleur.red(error));
+  if (callback && typeof callback === 'function') {
+    callback();
+  }
+  process.exit(exitCode);
+};
+
+const removeDirectory = () => {
+  rimraf(destination);
```
Comment (bergundy) [inline download-temporal-cli.js:30]: You should await this

File: download-temporal-cli.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+console.log(
+  kleur.bgYellow('Downloading:'),
+  kleur.yellow().underline(downloadUrl),
+);
+
+try {
+  const response = await fetch(downloadUrl);
+
+  await finished(
+    response.body.pipe(zlib.createGunzip()).pipe(tar.extract(destination)),
```
Comment (bergundy) [inline download-temporal-cli.js:55]: Are we not testing on windows?

**Pull request merged**